resources/js/entrypoints/chat.tsx

```diff
@@ -77,8 +81,10 @@ core.reactTurbolinks.register('chat', action(() => {
     core.dataStore.channelStore.lastReceivedMessageId = initial.last_message_id ?? 0;
   }
 
-  if (currentUrl().hash === '#create') {
-    core.dataStore.chatState.selectChannel('create', 'replaceHistory');
+  const hash = getChannelHash(currentUrl().hash);
+
+  if (hash === 'create' || hash === 'join') {
```

Comment:
is it any better than just comparing with `#create` and `#join`? Also `getChannelHash` is really just get hash without the hash sign, nothing to do with channel?

Reply:
it narrows the type down for `selectChannel`

Reply:
I see... ಠ_ಠ

The function name is still weird though and I don't think it's any better than straight out `.hash.slice(1)` since it must have the `#` by definition and clear enough in context (and slicing empty string doesn't do any harm).

Reply:
I just got rid of the function since it didn't end up being used for anything else.